### PR TITLE
Added logs to cluster extraction

### PR DESCRIPTION
**Pull request by rmweir**

### CODE REVIEW

File: controller/cluster_check.go [hunk 1]
```diff
@@ -116,3 +119,13 @@ func getFieldValue(v reflect.Value, name ...string) reflect.Value {
...
 }
+
+func debugLogsClusterExtraction(clusterName string, obj interface{}) {
+	if clusterName != "" {
+		return
+	}
+	ns := getValue(obj, "Namespace")
+	name := getValue(obj, "Name")
+	kind := getValue(obj, "Kind")
+	logrus.Debugf("failed to extract cluster that the object [%s:%s] of kind [%s] belongs to", ns, name, kind)
```
Comment (KevinJoiner) [inline controller/cluster_check.go:130]: nit: I fear that this makes it into production, and a user sees the logs and assumes that something in rancher is failing, causing them to raise a ticket.
```suggestion
	logrus.Debugf("unable to extract cluster that the object [%s:%s] of kind [%s] belongs to", ns, name, kind)
```

File: controller/cluster_check.go [hunk 2]
```diff
@@ -17,6 +17,8 @@ func ObjectInCluster(cluster string, obj interface{}) bool {
 	// Check if the object implements the interface, this is best case and
 	// what objects should strive to be
 	if o, ok := obj.(ObjectClusterName); ok {
+		clusterName := o.ObjClusterName()
+		debugLogsClusterExtraction(clusterName, obj)
 		return o.ObjClusterName() == cluster
```
Comment (KevinJoiner) [inline controller/cluster_check.go:22]:
```suggestion
		return clusterName == cluster
```

**Review by KevinJoiner — CHANGES_REQUESTED** [review]
Is the desire to leave this in forever? I am concerned that there are healthy Rancher setups that will start spamming this message since we are not sure what causes it. Or can we assume that anytime clusterName is empty, something is not correct?

**Comment by rmweir** [comment]
@KevinJoiner We are trying to figure out how often this happens. If it happens frequently then that is a problem. I don't think spam will be much of an issue since it's debug. The handler also have a built in backoff so it should be throttled.

**Review by cbron — APPROVED** [review]
lgtm, since it is debug only. Sort of sucks it will run for every case, but should be minimal.

**Review by KevinJoiner — APPROVED** [review]

**Pull request merged**